## Local regression tests failed

The implement subagent opened this PR even though the local `python -m unittest` run failed after commit. The failures below are posted here as a reviewer finding so `cai-revise` picks them up and addresses them on the PR side rather than rolling the issue back through another full implement cycle.

### Failing tests

```
Failing tests:
  - FAIL: test_cmd_helpers_git.TestReadPrefetchedFiles.test_issue_body_without_files_to_change_section
  - FAIL: test_cmd_helpers_git.TestReadPrefetchedFiles.test_no_issue_body_no_preload

FAIL: test_issue_body_without_files_to_change_section (test_cmd_helpers_git.TestReadPrefetchedFiles.test_issue_body_without_files_to_change_section)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/cai-implement-1224-5f22cf4c/tests/test_cmd_helpers_git.py", line 127, in test_issue_body_without_files_to_change_section
    self.assertNotIn("## Pre-loaded file contents", result)
AssertionError: '## Pre-loaded file contents' unexpectedly found in '## Work directory\n\nYou are running with cwd `/app` so your declarative agent definition and per-agent memory are read from the canonical image / volume locations. Your actual work happens on a fresh clone of the repository at:\n\n    /tmp/tmpwvw21ymw\n\n**You MUST use absolute paths under that directory for all Read/Edit/Write/Glob/Grep calls that target the work.** Relative paths resolve to `/app` (the canonical, baked-in version of the repo) which you should treat as read-only. Edits to `/app/...` would land in the container\'s writable layer and be lost on next restart — they would NOT make it into git.\n\nExamples:\n  - GOOD: `Read("/tmp/tmpwvw21ymw/cai.py")`\n  - BAD:  `Read("cai.py")`               (reads /app/cai.py)\n  - GOOD: `Edit("/tmp/tmpwvw21ymw/parse.py", ...)`\n  - BAD:  `Edit("parse.py", ...)`        (edits /app/parse.py)\n\nIf you have Bash in your tool allowlist, the same rule applies: use `git -C` (or absolute paths) for any git operation that should target the clone, NOT the cwd.\n\n  - GOOD: `git -C /tmp/tmpwvw21ymw status`\n  - BAD:  `git status`        (reports /app status, not the clone\'s)\n\n## Updating `.claude/agents/*.md` (self-modification)\n\nClaude-code\'s headless `-p` mode hardcodes a write block on every `.claude/agents/*.md` path, regardless of any permission flag or settings rule. Edit/Write calls against `/tmp/tmpwvw21ymw/.claude/agents/<name>.md` WILL fail with a sensitive-file protection error — this is not something you can bypass from inside your session.\n\nThe wrapper provides a **staging directory** at:\n\n    /tmp/tmpwvw21ymw/.cai-staging/agents\n\nTo update an `.claude/agents/<name>.md` file, use the Write tool to write the COMPLETE new file content (frontmatter + body) to the corresponding path under `/tmp/tmpwvw21ymw/.cai-staging/agents/`. After your session exits successfully the wrapper recursively walks every file it finds in `.cai-staging/agents/`, preserving subdirectory paths, and copies each one to the matching `.claude/agents/<relative-path>.md` in the clone, then deletes the staging directory so it never lands in the PR.\n\nRules:\n  - Staged files are copied unconditionally — new agent definitions are created if no target exists yet.\n  - Write the FULL file, not a diff or patch

... (truncated)
```

---
_Posted by `cai implement`. Addressing this comment clears the finding; the PR continues through review → merge as usual._

--- a/docs/modules/audit.md
+++ b/docs/modules/audit.md
@@ -10,9 +10,10 @@ function in `cai_lib/cmd_agents.py` or `cai_lib/cmd_misc.py`.
 
 ## Key entry points
 - [`cai_lib/audit/cost.py`](../../cai_lib/audit/cost.py) — `_load_outcome_counts`,
-  `_load_cost_log`, `_row_ts`, `_primary_model`,
-  `_build_cost_summary`; token/cost helpers consumed by
-  `cmd_cost_report` and `cmd_cost_optimize`.
+  `_load_cost_log`, `_load_outcome_index`, `_row_ts`, `_primary_model`,
+  `_build_module_index`, `_infer_module_from_files`, `_build_cost_summary`; 
+  token/cost helpers consumed by `cmd_cost_report`, `cmd_cost_optimize`, 
+  and `runner.py` for per-module audit context.
 - [`cai_lib/audit/modules.py`](../../cai_lib/audit/modules.py) —
   `ModuleEntry` dataclass; `load_modules(path, check_doc_exists)`
   and `coverage_check(modules, files)`. Drives
@@ -62,6 +63,15 @@ function in `cai_lib/cmd_agents.py` or `cai_lib/cmd_misc.py`.
   to `findings.json`.
 
 ## Operational notes
+- **Cost summary pre-load.** `_build_module_prompt` in `runner.py` now
+  calls `pull_cost()` (if sync is configured) to fetch fresh cost data 
+  before invoking each audit agent. `_build_cost_summary()` generates a 
+  rich 7-section markdown block (cost by agent, top targets, phase breakdown, 
+  per-module spend, cache-health regressions, host anomalies) and appends 
+  it as a `## Cost summary` section in the agent's user message. This 
+  provides spend context per module without the agent needing to parse 
+  log files. The cost summary is also printed to stderr when running 
+  interactively (TTY).
 - **Audit log path convention.** `cai_lib/audit/runner.py` writes one
   structured JSONL file per `(kind, module)` pair under
   `/var/log/cai/audit/<kind>/<module>.jsonl` (e.g.

--- a/tests/test_cmd_helpers_git.py
+++ b/tests/test_cmd_helpers_git.py
@@ -122,13 +122,20 @@ class TestReadPrefetchedFiles(unittest.TestCase):
     def test_issue_body_without_files_to_change_section(self):
         tmp = self._make_tmp()
         issue_body = "Some issue body without any files section."
-        result = _work_directory_block(tmp, issue_body)
+        # Suppress shared-memory injection (implement-plan-scope-gate.md
+        # contains the literal phrase "## Pre-loaded file contents" which
+        # would cause false-positive assertNotIn failures — same pattern
+        # as PR#1204 / PR#1226).
+        with patch("cai_lib.cmd_helpers_git._read_shared_memory", return_value=""):
+            result = _work_directory_block(tmp, issue_body)
         # Should fall back to current behavior — no preload section.
         self.assertNotIn("## Pre-loaded file contents", result)
 
     def test_no_issue_body_no_preload(self):
         tmp = self._make_tmp()
-        result = _work_directory_block(tmp)
+        # Same shared-memory isolation as test_issue_body_without_files_to_change_section.
+        with patch("cai_lib.cmd_helpers_git._read_shared_memory", return_value=""):
+            result = _work_directory_block(tmp)
         self.assertNotIn("## Pre-loaded file contents", result)
 
 

